src/tests/ftest/pool/dynamic_server_pool.py

```diff
@@ -0,0 +1,150 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from apricot import TestWithServers
+from check_for_pool import check_for_pool
+
+
+class DynamicServerPool(TestWithServers):
+    """Test Class Description:
+    Test to create pools on single, multiple, and dynamically added server.
+
+    This test covers the following requirements.
+    SRS-10-0054: New server can be used for new pool.
+    SRS-10-0105: A pool spanning only a subset of the servers can be created
+    starting from 1 server up to all the servers. Query should be used to verify
+    that the pool was allocated on the requested servers.
+
+    First, we'll create one pool on the first server and another pool over two
+    servers. This covers SRS-10-0105.
+
+    Second, we start a new server and let it join the existing two servers.
+    Then we'll create one pool on this new server and another one over all three
+    servers. This covers SRS-10-0054.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DynamicServerPool object."""
+        super(DynamicServerPool, self).__init__(*args, **kwargs)
+        self.expected_uuids = []
+
+    def setUp(self):
+        super(DynamicServerPool, self).setUp()
+        self.pool = []
+
+    def verify_uuids(self):
+        """Call dmg pool list to get the list of UUIDs and verify against the
+        expected UUIDs.
+        """
+        actual_uuids = []
+        output_list = self.get_dmg_command().get_output("pool_list")
+        for uuid_sr in output_list:
+            actual_uuids.append(uuid_sr[0])
+        actual_uuids.sort()
+        self.expected_uuids.sort()
+        self.assertEqual(self.expected_uuids, actual_uuids)
+
+    def check_pool_location(self, hosts, uuid_to_ranks):
+        """Iterate pools and verify that the UUID-named directory exists in
+        each hosts.
+
+        Args:
+            hosts (list of str): Hostnames to search the UUID-named directory.
+            uuid_to_ranks (str to list of int dictionary): UUID to rank list
+                dictionary.
+        """
+        RC_SUCCESS = 0
+        for pool in self.pool:
+            # Note that we don't check mapping between rank and hostname, but it
+            # appears that self.hostlist_servers[0] is always rank0, 1 is rank1,
+            # and the extra server we'll be adding will be rank2.
+            for rank, host in enumerate(hosts):
+                if(check_for_pool(host, pool.uuid.lower()) == RC_SUCCESS):
```

Comment:
(pylint-superfluous-parens) Unnecessary parens after 'if' keyword
(pylint-simplifiable-if-statement) The if statement can be replaced with 'var = bool(test)'